### PR TITLE
Re-implement the Billy

### PR DESCRIPTION
**Pull request by Garanas**
The Billy has an arm period of about 2.5 seconds. If the Billy collides with anything before it is armed then it does not explode. Instead, a little bit of debris that is on fire is spawned in to help the player to understand what happened.

The effects of the Billy are slightly adjusted to make it look a bit different from a regular strategic missile.

The Billy now provides vision where it explodes.

The outer damage radius of the Billy no longer pierces shields.

_Billy colliding before it is armed_

https://github.com/FAForever/fa/assets/15778155/4e197ad6-fe5f-4c83-ac1f-643ffec4810d

_When Billy is armed, a bit of debris that is on fire drops_

https://github.com/FAForever/fa/assets/15778155/5ab23661-ed23-4d61-acce-bf472f00b6f0

### CODE REVIEW

**Comment by lL1l1** [comment]
I wrote a PR for this PR with some bug fixes and reverting damage behavior: #5791 
I'll add here that keeping the damage in the BP is good for balancing and UI. Ahwassa at least doesn't modify its damage radius or total damage.

**Comment by Garanas** [comment]
Where does the current implementation change the total damage? The center receives full damage, as expected.

**Comment by Garanas** [comment]
Ah, I see now what you mean. Alright, let me fiddle with it a bit more.

**Comment by lL1l1** [comment]
Yes the center receives the same total damage, but billy nuke's splash damage is extremely important.
#### Before:
 0 to 15 radius: 12250 damage
15 to 25 radius: 250 damage
These values are easily displayed in the weapon details and the reticle with exact accuracy.
#### Now (converting from percents to numbers):
0 to 7.5 radius: 12500 damage (1)
7.5 to 15 radius: 8250 damage (2)
15 to 22.5 radius: 4500 damage (3)
22.5 to 25 radius: 750 damage (4)

(1) T3 Mobile shield (10000 HP) killing range reduced 50%.
(2) Can't kill shielded Seraphim armies (shields absorb all) or UEF armies (5250 damage to percies).
(3) Massive 7.5 range increase (+50%) for the significant damage amount. It would leave harbs/othuums at 100/200 HP, and remove over half the HP of bricks/percies. It would also kill t2 mobile shields.
(4) Can kill T2/T3 Engineers and Volcanoes now, not super important though.

The points about shields are made assuming that shields can't be layered to absorb many times more damage, i.e. multiple shields act as a single shield in terms of HP. This is not a bad assumption to make because from a balance perspective it's a desirable ability to maintain for Billy. Visually, it can be accomplished by having Billy reliably target the tops of mobile shield domes instead of flying into them from the side. That trajectory would also be good for the player to have a reliable target point for the nuke. Coding-wise, from my testing, it would most likely need a custom damage function against specifically mobile shields.

For UI, it would be very difficult to cram all that information into the weapon details. The reticle would also be inconsistent with others because it can't display 4 layers of damage, and the player won't have information on where they can deal 8250 damage or 4500 damage.

**Comment by lL1l1** [comment]
Also the minimum radius increase doesn't show up on the range rings, it should probably be reverted since the damage was reverted too, unless there's a way to disable one weapon's range and display the other's.

**Comment by Garanas** [comment]
The minimum radius increase is there to help the player understand that the weapon doesn't work up close (because it is not armed). It has the same range as the omni ring, not optimal of course as it would be better if it had a separate ring. Not sure why it does not show

**Comment by lL1l1** [comment]
It doesn't show because it's in the same range category as the TML which has a shorter minimum range, which is why I suggested for a way to disable one weapon's range ring and show the other.

**Comment by Basilisk3** [comment]
This looks promising👍.

I think it makes sense if the explosion can pierce mobile shields like a regular nuke, but cannot pierce static shields, as it is only a tactical one. If it is implemented like this, that would have to be communicated in the UI, though. Maybe it could be part of the upgrade's description? The damage values could be added there too.

**Comment by lL1l1** [comment]
It should not explode when impacting Volcano flares: https://github.com/FAForever/fa/commit/8fb229dfa0891c813f184cc95810e67cfc6a72dc could be placed in the projectile script or the TIFTacticalNuke class.
The shield behavior is okay, it should have never bypassed shields, and that was a change caused by the tactical missile rework: https://github.com/FAForever/fa/commit/8db5c4620b6b4ddd04687f2def2995795314428b#diff-e0e266efde489a9c85bbd043f6448d0a088454b121cf2f05e089927eace970bbL39-L48

**Comment by Garanas** [comment]
Great work @lL1l1 , thank you for the feedback on everything

File: units/UEL0001/UEL0001_unit.bp [hunk 1]
```diff
@@ -1041,7 +1042,7 @@ UnitBlueprint{
             ManualFire = true,
             MaxProjectileStorage = 1,
             MaxRadius = 256,
-            MinRadius = 15,
+            MinRadius = 26,
```
[the remark this answers is not in the document]
Comment (Garanas) [reply units/UEL0001/UEL0001_unit.bp:1045]: The weapon is properly disabled by script.

It appears just the existence of a weapon with a larger minimum radius than the maximum radius of the 'primary' weapon causes this behavior.

**Pull request merged**